generators/utilities/src/main/resources/csrc/emulator.cc

```diff
@@ -325,6 +336,13 @@ int main(int argc, char** argv)
 #endif
     trace_count++;
   }
+  // verilator multithreading. need to do 1 loop before checking if
```

Comment:
Did TSI not work in verilator before? Why does the multithreading affect this?

Reply:
no it didn't work before, because the main thread created the host TSI object, whose `cur` context is stored in local thread storage. when a different thread calls SimSerial's tick(), it does not have access to the TSI object's `cur` since the context object is local to the main thread (see https://github.com/riscv/riscv-isa-sim/blob/667433bbee4ba0a10f569ac4ad8b52578574ff31/fesvr/context.cc#L6) 

if we allow the simulator to do 1 loop, we can be sure the global TSI object was created by the simulator thread that will be calling tick()

Reply:
Just for my own clarity, when you say it didn't work before- do you mean that it literally didn't work, or that it was a multithreading bug that was masked because everything was always single-threaded?

Reply:
i mean, it literally didn't work under any condition when multithreading was enabled.

Reply:
the code was functionally correct when 1 thread was used though